File: cpp/1980.Find-Unique_Binary_String.cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <string>
#include <vector>
class Solution {
public:
  std::string findDifferentBinaryString(std::vector<std::string> &nums) {
    int n = nums.size();

    std::string result;
    for (int i = 0; i < n; i++) {
      result += nums[i][i] == '0' ? '1' : '0';
    }

    return result;
  }
};

File: python/1980.find-unique-binary-string.py
from typing import List


class Solution:
    # len(nums) is expected to be equal to len(nums[i])
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        n = len(nums)
        set = ()

        seen = [int(num, 2) for num in nums]

        # Base on Pigeonhole Principle, we only have to
        # enumerate until len(nums)+1 instead of limit(n)+1
        for num in range(n + 1):
            if num not in seen:
                return f"{num:0{n}b}"

        return ""
